python/ga4gh/dos/schema.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+import os.path
+
+import swagger_spec_validator.common
+
+cd = os.path.dirname(os.path.realpath(__file__))
+SWAGGER_PATH = os.path.join(cd, 'data_object_service.swagger.yaml')
+
+
+def present_schema():
+    """
+    Presents the OpenAPI 2.0 schema as a dictionary.
+    :rvtype: dict
+    """
+    return swagger_spec_validator.common.read_file(SWAGGER_PATH)
+
+
+def from_chalice_routes(routes, base_path=''):
+    """
+    Given a :obj:`chalice.Chalice.routes` objects, computes the proper
+    subset of the Data Object Service schema and presents it as an
+    OpenAPI 2.0 JSON schema.
+    :rvtype: dict
+    :param chalice.Chalice.routes routes:
+    :param str base_path: the base path of the endpoints listed in `routes`.
+                          This is only necessary if a base path is manually
+                          prepended to each endpoint your service exposes,
+                          e.g. ``@app.route('/ga4gh/dos/v1/dataobjects')``.
```

Comment:
This is really great work. It allows one to say, "I made a service with some DOS endpoints" and then generate a swagger doc just for those endpoints that we have documented. This means that implementations in chalice that follow the schema's routes can have documentation generated for them automatically.